src/nft/components/bag/BagFooter.tsx

```diff
@@ -401,14 +412,15 @@ export const BagFooter = ({ setModalIsOpen, eventProperties }: BagFooterProps) =
       return getBuyButtonStateData(BuyButtonStates.FETCHING_TOKEN_ROUTE, theme)
     }
 
-    if (allowance.state === AllowanceState.REQUIRED || loadingAllowance) {
-      const handleClick = () => updateAllowance()
+    const allowanceRequired = allowance.state === AllowanceState.REQUIRED
+    if (allowanceRequired || loadingAllowance) {
+      const handleClick = () => allowanceRequired && allowance.approveAndPermit()
 
       if (loadingAllowance) {
         return getBuyButtonStateData(BuyButtonStates.LOADING_ALLOWANCE, theme, handleClick)
-      } else if (isAllowancePending) {
+      } else if (allowanceRequired && allowance.isApprovalPending) {
```

Comment:
we actually do to make the type checker happy

Reply:
The linter doesn't seem to parse that since you already checked `allowanceRequired || loadingAllowance` and confirmed that `loadingAllowance` is false, this must mean that `allowanceRequired` is true. 

Looking over the code @JackShort it doesn't seem like there's that much overlap between allowanceRequired and loadingAllowance, notably the `onClick` doesn't affect the button when it's in the `LOADING_ALLOWANCE` state. Could we split the code as follows? 

```
    const allowanceRequired = allowance.state === AllowanceState.REQUIRED
    if (loadingAllowance) {
      return getBuyButtonStateData(BuyButtonStates.LOADING_ALLOWANCE, theme, handleClick)
    }

    if (allowanceRequired) {
      const handleClick = () => allowance.approveAndPermit()

      if (allowance.isApprovalPending) {
        return getBuyButtonStateData(BuyButtonStates.IN_WALLET_ALLOWANCE_APPROVAL, theme, handleClick)
      } else if (allowance.isApprovalLoading) {
        return getBuyButtonStateData(BuyButtonStates.PROCESSING_APPROVAL, theme, handleClick)
      }

      return getBuyButtonStateData(BuyButtonStates.REQUIRE_APPROVAL, theme, handleClick)
    }